src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,295 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Component, Directive, Input, Output,
+  OnDestroy, AfterViewInit, ElementRef, Injectable, Optional} from '@angular/core';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+
+
+/**
+ * 'cdkStickyRegion' is a region contains one sticky header and other
+ * contents that user wants to put under the sticky-header. There only can
+ * be one sticky-header in one 'cdkStickyRegion'.
+ * If a user does not define a 'cdkStickyRegion' for a sticky-header, the direct
+ * parent node of the sticky-header will be set as the 'cdkStickyRegion'.
+ */
+@Directive({
+  selector: '[cdkStickyRegion]',
+})
+export class CdkStickyRegion {
+  constructor(private _element: ElementRef) { }
+
+  getElementRef(): ElementRef {
+    return this._element;
+  }
+}
+
+
+/**
+ * The 'cdkStickyHeader' is the header which user wants to be stuck at top of the
+ * scrollable container. The main logic is in the 'sticker()' function.
+ */
+const STICK_START_CLASS = 'stick-start';
+const STICK_END_CLASS = 'sticky-end';
```

Comment:
Should I change it into 'mat_STICK_START_CLASS' or 'MAT_STICK_START_CLASS' or 'mat_stick_start_class'?
I'm currently change it into 'mat_STICK_START_CLASS', ..which looks a little bit strange...